src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +70,812 @@ const ActorInit En_Fz_InitVars = {
+    ICHAIN_F32(unk_4C, 30, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fz/func_80A20EC4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fz/EnFz_Update.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fz/EnFz_Draw.s")
+void EnFz_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFz* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fz/func_80A2145C.s")
+    Actor_ProcessInitChain(&this->actor, &sInitChain);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fz/func_80A21500.s")
+    this->actor.colChkInfo.damageTable = &sDamageTable;
+    this->actor.colChkInfo.health = 6;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fz/func_80A215BC.s")
+    Collider_InitCylinder(globalCtx, &this->collider1);
+    Collider_SetCylinder_Set3(globalCtx, &this->collider1, &this->actor, &sCylinderInit1);
+
+    Collider_InitCylinder(globalCtx, &this->collider2);
+    Collider_SetCylinder_Set3(globalCtx, &this->collider2, &this->actor, &sCylinderInit2);
+
+    Collider_InitCylinder(globalCtx, &this->collider3);
+    Collider_SetCylinder_Set3(globalCtx, &this->collider3, &this->actor, &sCylinderInit3);
+    Actor_SetScale(&this->actor, 0.008f);
+
+    this->actor.colChkInfo.mass = 0xFF;
+
+    this->actor.flags &= ~1;
+
+    this->unusedTimer1 = 0;
+    this->unusedCounter = 0;
+    this->isAlwaysTrue = true;
+    this->isMoving = false;
+    this->isFreezing = false;
+    this->isActive = true;
+    this->isDespawning = false;
+
+    this->actor.speedXZ = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->actor.velocity.y = 0.0f;
+
+    this->posOrigin.y = this->actor.posRot.pos.y;
+    this->IceSmokeFreezingSpawnHeight = this->actor.posRot.pos.y;
+
+    this->posOrigin.x = this->actor.posRot.pos.x;
+    this->posOrigin.z = this->actor.posRot.pos.z;
+
+    this->unusedFloat = 135.0f;
+
+    if (this->actor.params < 0) {
+        this->envAlpha = 0;
+        this->actor.scale.y = 0.0f;
+        EnFz_Hidden(this);
+    } else {
+        this->envAlpha = 255;
+        EnFz_BeginFreezingStationary(this);
+    }
+
+    EnFz_GetCollision(this, globalCtx);
+}
+
+void EnFz_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFz* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyCylinder(globalCtx, &this->collider2);
+    Collider_DestroyCylinder(globalCtx, &this->collider3);
+}
+
+void EnFz_GetCollision(EnFz* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f collisionPos;
+    Vec3f vec1;
+    u32 bgId;
+    CollisionPoly* poly;
+
+    pos.x = this->actor.posRot.pos.x;
+    pos.y = this->actor.posRot.pos.y + 20.0f;
+    pos.z = this->actor.posRot.pos.z;
+    Matrix_Translate(pos.x, pos.y, pos.z, MTXMODE_NEW);
+    Matrix_RotateRPY(this->actor.shape.rot.x, this->actor.shape.rot.y, this->actor.shape.rot.z, MTXMODE_APPLY);
+
+    vec1.y = 0.0f;
+    vec1.x = 0.0f;
```

Comment:
```suggestion
   vec1.x = vec1.y = 0.0f;
```
Vectors are usually assigned in xyz order, so when they're out of order like this it's often a chained assignment.